include/nbl/asset/utils/IShaderCompiler.h

```diff
@@ -13,15 +13,16 @@
 #include "nbl/asset/ICPUShader.h"
 #include "nbl/asset/utils/ISPIRVOptimizer.h"
 
+#include "nbl/core/xxHash256.h"
+
+#include <nlohmann/json.hpp>
```

Comment:
do you need to leak this to public headers?

Reply:
you can forward declare `nlohmann::json` instead btw just a
```cpp
namespace nlohmann
{
class json;
}
```

Reply:
What about the hashing? Do I move it to the .cpp?

Reply:
> What about the hashing? Do I move it to the .cpp?

yes, also while you are on it please do also [this](https://github.com/Devsh-Graphics-Programming/Nabla/pull/676/files#r1580487872)

Reply:
one more thing

```c++
#include "nlohmann/json.hpp"
```

not

```c++
#include <nlohmann/json.hpp>
```

Reply:
> > What about the hashing? Do I move it to the .cpp?
> 
> yes, also while you are on it please do also [this](https://github.com/Devsh-Graphics-Programming/Nabla/pull/676/files#r1580487872)

no stays where it is, was supposed to be a public header anyway.

Reply:
> > > What about the hashing? Do I move it to the .cpp?
> > 
> > 
> > yes, also while you are on it please do also [this](https://github.com/Devsh-Graphics-Programming/Nabla/pull/676/files#r1580487872)
> 
> no stays where it is, was supposed to be a public header anyway.

I meant to move `#include` part to .cpp to not expose headers, sry

Reply:
@AnastaZIuk review if fixed

Reply:
> one more thing
> 
> ```c++
> #include "nlohmann/json.hpp"
> ```
> 
> not
> 
> ```c++
> #include <nlohmann/json.hpp>
> ```

still needs fixing

Reply:
If I forward declare `nlohmann::json` I get `nlohmann::json is ambiguous` compilation errors, even if I don't include the `json.h` here.  As for the hashing, since it's constexpr there is no forwarding, so I'm moving anything that hashes out of the .h and into the .cpp

Reply:
btw the reason it was ambiguous is because `nlohmann::json` was a `using` alias, and not a struct or class, fixing by myself.